lib/DateTime/Format/Strptime.pm

```diff
@@ -426,7 +426,7 @@ sub _build_parser {
             field => 'year',
         },
         z => {
-            regex => qr/[+-]$d{4}/,
+            regex => qr/(?: [Z] | [+-][0-9]{2} (?: [:]? [0-9]{2})? )/x,
```

Comment:
There's a `$digit` you can use instead of `[0-9]`. Also, while I like the `[Z]` thing the existing regexes are not using that or `/x`, and I'd prefer consistency here. A separate PR to make all the regexes prettier would be cool.